### DISCUSSION TITLE
checkoutLinesAdd by anonymous user without authorization

### DISCUSSION THREAD

**Discussion by tanujrawat95** [saleor/saleor#12119] [Q&A]
Hi everyone!
I created a checkout with mutation CreateCheckout with bearer token as header, so that checkout gets created for the authenticated user linked to the token. It is working as expected.

If I want to fetch the created checkout, I need to pass the bearer token as the checkout is binded to a specific user, which is absolutely valid and working. No guest/other users can access the checkout.

But when i try to add lines to the same cart with mutation checkoutLinesAdd, it is working even if i dont send access token.
This means anyone can make changes to any other cart weather authenticated or not.

I am using SSO implementation with Keycloak.

I have already raised an issue. But wanted to know if the current approach is valid or not?

**Comment by maarcingebala** [comment]
Hi @tanujrawat95, thanks for raising this. Which version of Saleor are you using? We will confirm that issue and get back soon.

**Reply by tanujrawat95** [reply]
Hi @maarcingebala, I am using the latest build of saleor platform which is using docker image "ghcr.io/saleor/saleor:3.11.8"

**Reply by maarcingebala** [reply]
Thanks, I confirmed the issue, we'll add it to our backlog to fix it.

**Reply by maarcingebala** [reply]
Link to issue for reference: https://github.com/saleor/saleor/issues/12110
